WordPress/Sniff.php

```diff
@@ -2645,4 +2645,38 @@ public function is_use_of_global_constant( $stackPtr ) {
 		return true;
 	}
 
+	/**
+	 * Determine if the parameter has a falsy value.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param array $parameter Array with information about the parameter.
+	 *
+	 * @return bool True if the parameter is falsy.
+	 */
+	protected function is_falsy_parameter( $parameter ) {
+		$falseTokens = Tokens::$emptyTokens;
+		$falseTokens[] = T_NULL;
+		$falseTokens[] = T_FALSE;
+
+		// Find anything excluding the false tokens.
+		$hasNonFalse = $this->phpcsFile->findNext( $falseTokens, $parameter['start'], ( $parameter['end'] + 1 ), true );
+		// If no false tokens are found, we are good.
+		if ( false === $hasNonFalse ) {
+			return true;
+		}
+
+		// Find out it has a number value.
+		$numberToken = $this->phpcsFile->findNext( T_LNUMBER, $parameter['start'], ( $parameter['end'] + 1 ) );
+		if ( $numberToken ) {
+			// If the number is 0, it will evaluate to false and needs to be caught.
```

Comment:
The `T_LNUMBER`, according to https://secure.php.net/manual/en/tokens.php, can be values in multiple bases - i.e. `0x13`. Presumably, this could also be something equating to 0, like `0x000`? If so, then the identity check `=== '0'` below will incorrectly not match.

Reply:
Put another way, could it be a string like `'000'`?